<?xml version="1.0" ?>
<robot name="panda" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <link name="panda_link0">
    <visual>
    <geometry>
	    <mesh filename="file:///opt/ros/humble/share/moveit_resources_panda_description/meshes/visual/link0.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
	      <mesh filename="file:///opt/ros/humble/share/moveit_resources_panda_description/meshes/collision/link0.stl"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="2.8142028896"/>
      <inertia ixx="0.0129886979" ixy="0.0" ixz="0.0" iyy="0.0165355284" iyz="0" izz="0.0203311636"/>
    </inertial>
  </link>
  <link name="panda_link1">
    <visual>
      <geometry>
	      <mesh filename="file:///opt/ros/humble/share/moveit_resources_panda_description/meshes/visual/link1.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
	      <mesh filename="file:///opt/ros/humble/share/moveit_resources_panda_description/meshes/collision/link1.stl"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="2.3599995791"/>
      <inertia ixx="0.0186863903" ixy="0.0" ixz="0.0" iyy="0.0143789874" iyz="0" izz="0.00906812"/>
    </inertial>
  </link>
  <joint name="panda_joint1" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.8973" soft_upper_limit="2.8973"/>
    <origin rpy="0 0 0" xyz="0 0 0.333"/>
    <parent link="panda_link0"/>
    <child link="panda_link1"/>
    <axis xyz="0 0 1"/>
    <limit effort="87" lower="-2.8973" upper="2.8973" velocity="2.1750"/>
  </joint>
  <link name="panda_link2">
    <visual>
      <geometry>
	      <mesh filename="file:///opt/ros/humble/share/moveit_resources_panda_description/meshes/visual/link2.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="file:///opt/ros/humble/share/moveit_resources_panda_description/meshes/collision/link2.stl"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="2.379518833"/>
      <inertia ixx="0.0190388734" ixy="0.0" ixz="0.0" iyy="0.0091429124" iyz="0" izz="0.014697537"/>
    </inertial>
  </link>
  <joint name="panda_joint2" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-1.7628" soft_upper_limit="1.7628"/>
    <origin rpy="-1.57079632679 0 0" xyz="0 0 0"/>
    <parent link="panda_link1"/>
    <child link="panda_link2"/>
    <axis xyz="0 0 1"/>
    <limit effort="87" lower="-1.7628" upper="1.7628" velocity="2.1750"/>
  </joint>
  <link name="panda_link3">
    <visual>
      <geometry>
        <mesh filename="file:///opt/ros/humble/share/moveit_resources_panda_description/meshes/visual/link3.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="file:///opt/ros/humble/share/moveit_resources_panda_description/meshes/collision/link3.stl"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="2.6498823337"/>
      <inertia ixx="0.0129300178" ixy="0.0" ixz="0.0" iyy="0.0150242121" iyz="0" izz="0.0142734598"/>
    </inertial>
  </link>
  <joint name="panda_joint3" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.8973" soft_upper_limit="2.8973"/>
    <origin rpy="1.57079632679 0 0" xyz="0 -0.316 0"/>
    <parent link="panda_link2"/>
    <child link="panda_link3"/>
    <axis xyz="0 0 1"/>
    <limit effort="87" lower="-2.8973" upper="2.8973" velocity="2.1750"/>
  </joint>
  <link name="panda_link4">
    <visual>
      <geometry>
        <mesh filename="file:///opt/ros/humble/share/moveit_resources_panda_description/meshes/visual/link4.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="file:///opt/ros/humble/share/moveit_resources_panda_description/meshes/collision/link4.stl"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="2.6948018744"/>
      <inertia ixx="0.0133874611" ixy="0.0" ixz="0.0" iyy="0.014514325" iyz="0" izz="0.0155175551"/>
    </inertial>
  </link>
  <joint name="panda_joint4" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-3.0718" soft_upper_limit="-0.0698"/>
    <origin rpy="1.57079632679 0 0" xyz="0.0825 0 0"/>
    <parent link="panda_link3"/>
    <child link="panda_link4"/>
    <axis xyz="0 0 1"/>
    <limit effort="87" lower="-3.0718" upper="-0.0698" velocity="2.1750"/>
  </joint>
  <link name="panda_link5">
    <visual>
      <geometry>
        <mesh filename="file:///opt/ros/humble/share/moveit_resources_panda_description/meshes/visual/link5.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="file:///opt/ros/humble/share/moveit_resources_panda_description/meshes/collision/link5.stl"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="2.9812816864"/>
      <inertia ixx="0.0325565705" ixy="0.0" ixz="0.0" iyy="0.0270660472" iyz="0" izz="0.0115023375"/>
    </inertial>
  </link>
  <joint name="panda_joint5" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.8973" soft_upper_limit="2.8973"/>
    <origin rpy="-1.57079632679 0 0" xyz="-0.0825 0.384 0"/>
    <parent link="panda_link4"/>
    <child link="panda_link5"/>
    <axis xyz="0 0 1"/>
    <limit effort="12" lower="-2.8973" upper="2.8973" velocity="2.6100"/>
  </joint>
  <link name="panda_link6">
    <visual>
      <geometry>
        <mesh filename="file:///opt/ros/humble/share/moveit_resources_panda_description/meshes/visual/link6.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="file:///opt/ros/humble/share/moveit_resources_panda_description/meshes/collision/link6.stl"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="1.1285806309"/>
      <inertia ixx="0.0026052565" ixy="0.0" ixz="0.0" iyy="0.0039897229" iyz="0" izz="0.0047048591"/>
    </inertial>
  </link>
  <joint name="panda_joint6" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-0.0175" soft_upper_limit="3.7525"/>
    <origin rpy="1.57079632679 -0.5 0.0" xyz="0 0 0"/>
    <parent link="panda_link5"/>
    <child link="panda_link6"/>
    <axis xyz="0 0 1"/>
    <limit effort="12" lower="-0.0175" upper="3.7525" velocity="2.6100"/>
  </joint>
  <link name="panda_link7">
    <visual>
      <geometry>
        <mesh filename="file:///opt/ros/humble/share/moveit_resources_panda_description/meshes/visual/link7.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="file:///opt/ros/humble/share/moveit_resources_panda_description/meshes/collision/link7.stl"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.4052912465"/>
      <inertia ixx="0.0006316592" ixy="0.0" ixz="0.0" iyy="0.0006319639" iyz="0" izz="0.0010607721"/>
    </inertial>
  </link>
  <joint name="panda_joint7" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.8973" soft_upper_limit="2.8973"/>
    <origin rpy="1.57079632679 0 0" xyz="0.088 0 0"/>
    <parent link="panda_link6"/>
    <child link="panda_link7"/>
    <axis xyz="0 0 1"/>
    <limit effort="12" lower="-2.8973" upper="2.8973" velocity="2.6100"/>
  </joint>
  <link name="panda_link8"/>
  <joint name="panda_joint8" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.107"/>
    <parent link="panda_link7"/>
    <child link="panda_link8"/>
    <axis xyz="0 0 0"/>
  </joint>

    <joint name="R_gripper_base_joint" type="fixed">
      <origin
      xyz="0.04 0.04 0.04"
      rpy="3.1415926 0 0.0" />
      <parent link="panda_link8"/>
      <child link="R_flange"/>
    </joint>

    <link name="R_flange">
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/flange.STL" scale="0.001 0.001 0.001"/>
        </geometry>
        <material
          name="">
          <color
            rgba="1 1 1 1" />
        </material>
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/flange.STL" scale="0.001 0.001 0.001"/>
        </geometry>
      </collision>
      <inertial>
        <origin
          xyz="0 0 0"
          rpy="0 0 0" />
        <mass
          value="1.0372398E-01" />
        <inertia
          ixx="8.6985265E-05"
          ixy="0"
          ixz="2.7728445E-07"
          iyy="8.7567207E-05"
          iyz="6.8674339E-07"
          izz="7.4487068E-05"/>
      </inertial>
    </link>

    <joint 
      name="R_base_link_joint" 
      type="fixed">
      <origin
      xyz="0 0 0"
      rpy="1.570796 0 0" />
      <parent link="R_flange"/>
      <child link="R_hand_base_link"/>
    </joint>

  <link
    name="R_hand_base_link">
    <inertial>
      <origin
        xyz="-0.0025264 -0.066047 0.0019598"
        rpy="0 0 0" />
      <mass
        value="0.14143" />
      <inertia
        ixx="0.00012281"
        ixy="2.1711E-06"
        ixz="1.7709E-06"
        iyy="8.3832E-05"
        iyz="-1.6551E-06"
        izz="7.6663E-05" />
    </inertial>
    <visual>
      <geometry>
        <mesh
          filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh
          filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/base_link.STL" />
      </geometry>
    </collision>
  </link>
  <link
    name="R_thumb_proximal_base">
    <inertial>
      <origin
        xyz="-0.0048064 0.0009382 -0.00757"
        rpy="0 0 0" />
      <mass
        value="0.0018869" />
      <inertia
        ixx="5.816E-08"
        ixy="1.4539E-08"
        ixz="4.491E-09"
        iyy="7.9161E-08"
        iyz="-1.8727E-09"
        izz="6.7433E-08" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/Link11.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/Link11.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="R_thumb_proximal_yaw_joint"
    type="revolute">
    <origin
      xyz="-0.01696 -0.0691 -0.02045"
      rpy="1.5708 -1.5708 0" />
    <parent
      link="R_hand_base_link" />
    <child
      link="R_thumb_proximal_base" />
    <axis
      xyz="0 0 -1" />
    <limit
      lower="-0.1"
      upper="1.3"
      effort="10"
      velocity="0.5" />
  </joint>
  <link
    name="R_thumb_proximal">
    <inertial>
      <origin
        xyz="0.021932 0.012785 -0.0080386"
        rpy="0 0 0" />
      <mass
        value="0.0066075" />
      <inertia
        ixx="1.5686E-06"
        ixy="-7.8296E-07"
        ixz="8.9143E-10"
        iyy="1.7353E-06"
        iyz="-1.0191E-09"
        izz="2.786E-06" />
    </inertial>
    <visual>
      <geometry>
        <mesh
          filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/Link12.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/Link12.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="R_thumb_proximal_pitch_joint"
    type="revolute">
    <origin
      xyz="-0.0088099 0.010892 -0.00925"
      rpy="1.5708 0 2.8587" />
    <parent
      link="R_thumb_proximal_base" />
    <child
      link="R_thumb_proximal" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-0.1"
      upper="0.6"
      effort="10"
      velocity="0.5" />
  </joint>
  <link
    name="R_thumb_intermediate">
    <inertial>
      <origin
        xyz="0.0095544 -0.0016282 -0.0071997"
        rpy="0 0 0" />
      <mass
        value="0.0037847" />
      <inertia
        ixx="3.6981E-07"
        ixy="-9.8581E-08"
        ixz="-4.7469E-12"
        iyy="3.2394E-07"
        iyz="1.0939E-12"
        izz="4.6531E-07" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/Link13.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/Link13.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="R_thumb_intermediate_joint"
    type="revolute">
    <origin
      xyz="0.04407 0.034553 -0.0008"
      rpy="0 0 0" />
    <parent
      link="R_thumb_proximal" />
    <child
      link="R_thumb_intermediate" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="0"
      upper="0.8"
      effort="10"
      velocity="0.5" />
    <mimic 
      joint="R_thumb_proximal_pitch_joint"
      multiplier="1.6"
      offset="0" />
  </joint>
  <link
    name="R_thumb_distal">
    <inertial>
      <origin
        xyz="0.0092888 0.0049529 -0.0060033"
        rpy="0 0 0" />
      <mass
        value="0.0033441" />
      <inertia
        ixx="1.3632E-07"
        ixy="-5.6788E-08"
        ixz="-9.2764E-11"
        iyy="1.4052E-07"
        iyz="-1.2283E-10"
        izz="2.0026E-07" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/Link14.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/Link14.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="R_thumb_distal_joint"
    type="revolute">
    <origin
      xyz="0.020248 0.010156 -0.0012"
      rpy="0 0 0" />
    <parent
      link="R_thumb_intermediate" />
    <child
      link="R_thumb_distal" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="0"
      upper="1.2"
      effort="10"
      velocity="0.5" />
    <mimic 
      joint="R_thumb_proximal_pitch_joint"
      multiplier="2.4"
      offset="0" />
  </joint>
  <link
    name="R_index_proximal">
    <inertial>
      <origin
        xyz="0.0012259 0.011942 -0.0060001"
        rpy="0 0 0" />
      <mass
        value="0.0042403" />
      <inertia
        ixx="6.6232E-07"
        ixy="-1.5775E-08"
        ixz="1.8515E-12"
        iyy="2.1146E-07"
        iyz="-5.0828E-12"
        izz="6.9398E-07" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/Link15.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/Link15.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="R_index_proximal_joint"
    type="revolute">
    <origin
      xyz="0.00028533 -0.13653 -0.032268"
      rpy="-3.1067 0 0" />
    <parent
      link="R_hand_base_link" />
    <child
      link="R_index_proximal" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="0"
      upper="1.7"
      effort="10"
      velocity="0.5" />
  </joint>
  <link
    name="R_index_intermediate">
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/Link16.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/Link16.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="R_index_intermediate_joint"
    type="revolute">
    <origin
      xyz="-0.0026138 0.032026 -0.001"
      rpy="0 0 0" />
    <parent
      link="R_index_proximal" />
    <child
      link="R_index_intermediate" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="0"
      upper="1.7"
      effort="10"
      velocity="0.5" />
    <mimic 
      joint="R_index_proximal_joint"
      multiplier="1"
      offset="0" />
  </joint>
  <link
    name="R_middle_proximal">
    <inertial>
      <origin
        xyz="0.001297 0.011934 -0.0060001"
        rpy="0 0 0" />
      <mass
        value="0.0042403" />
      <inertia
        ixx="6.6211E-07"
        ixy="-1.8461E-08"
        ixz="1.8002E-12"
        iyy="2.1167E-07"
        iyz="-6.6808E-12"
        izz="6.9397E-07" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/Link17.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/Link17.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="R_middle_proximal_joint"
    type="revolute">
    <origin
      xyz="0.00028533 -0.1371 -0.01295"
      rpy="-3.1416 0 0" />
    <parent
      link="R_hand_base_link" />
    <child
      link="R_middle_proximal" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="0"
      upper="1.7"
      effort="10"
      velocity="0.5" />
  </joint>
  <link
    name="R_middle_intermediate">
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/Link18.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/Link18.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="R_middle_intermediate_joint"
    type="revolute">
    <origin
      xyz="-0.0024229 0.032041 -0.001"
      rpy="0 0 0" />
    <parent
      link="R_middle_proximal" />
    <child
      link="R_middle_intermediate" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="0"
      upper="1.7"
      effort="10"
      velocity="0.5" />
    <mimic 
      joint="R_middle_proximal_joint"
      multiplier="1"
      offset="0" />
  </joint>
  <link
    name="R_ring_proximal">
    <inertial>
      <origin
        xyz="0.001297 0.011934 -0.0060002"
        rpy="0 0 0" />
      <mass
        value="0.0042403" />
      <inertia
        ixx="6.6211E-07"
        ixy="-1.8461E-08"
        ixz="1.5793E-12"
        iyy="2.1167E-07"
        iyz="-6.6868E-12"
        izz="6.9397E-07" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/Link19.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/Link19.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="R_ring_proximal_joint"
    type="revolute">
    <origin
      xyz="0.00028533 -0.13691 0.0062872"
      rpy="3.0892 0 0" />
    <parent
      link="R_hand_base_link" />
    <child
      link="R_ring_proximal" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="0"
      upper="1.7"
      effort="10"
      velocity="0.5" />
  </joint>
  <link
    name="R_ring_intermediate">
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/Link20.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/Link20.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="R_ring_intermediate_joint"
    type="revolute">
    <origin
      xyz="-0.0024229 0.032041 -0.001"
      rpy="0 0 0" />
    <parent
      link="R_ring_proximal" />
    <child
      link="R_ring_intermediate" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="0"
      upper="1.7"
      effort="10"
      velocity="0.5" />
    <mimic 
      joint="R_ring_proximal_joint"
      multiplier="1"
      offset="0" />
  </joint>
  <link
    name="R_pinky_proximal">
    <inertial>
      <origin
        xyz="0.001297 0.011934 -0.0060001"
        rpy="0 0 0" />
      <mass
        value="0.0042403" />
      <inertia
        ixx="6.6211E-07"
        ixy="-1.8461E-08"
        ixz="1.6907E-12"
        iyy="2.1167E-07"
        iyz="-6.9334E-12"
        izz="6.9397E-07" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/Link21.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/Link21.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="R_pinky_proximal_joint"
    type="revolute">
    <origin
      xyz="0.00028533 -0.13571 0.025488"
      rpy="3.0369 0 0" />
    <parent
      link="R_hand_base_link" />
    <child
      link="R_pinky_proximal" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="0"
      upper="1.7"
      effort="10"
      velocity="0.5" />
  </joint>
  <link
    name="R_pinky_intermediate">
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/Link22.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="file:///home/student/Documents/tang/VisionRobot/nv8arm/foxy/nv6_ws/src/moveit_resources_rrr_description/meshes_inspire/Link22.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="R_pinky_intermediate_joint"
    type="revolute">
    <origin
      xyz="-0.0024229 0.032041 -0.001"
      rpy="0 0 0" />
    <parent
      link="R_pinky_proximal" />
    <child
      link="R_pinky_intermediate" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="0"
      upper="1.7"
      effort="10"
      velocity="0.5" />
    <mimic 
      joint="R_pinky_proximal_joint"
      multiplier="1"
      offset="0" />
  </joint>
</robot>
